nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb

```diff
@@ -31,6 +31,51 @@ sealed abstract class Array[T]
   protected override def clone(): Array[T] = undefined
 }
 
+// note: Array[Ptr[_]] is implemented as a ObjectArray
+
+final class ObjectArray private () extends Array[Object] {
```

Comment:
The duplication of code is unfortunate, but there is no easy way to fix it at the moment.